staging/src/k8s.io/kube-scheduler/config/v1alpha2/types_pluginargs.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/kube-scheduler/config/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// InterPodAffinityArgs holds arguments used to configure the InterPodAffinity plugin.
+type InterPodAffinityArgs struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// HardPodAffinityWeight is the scoring weight for existing pods with a
+	// matching hard affinity to the incoming pod.
+	HardPodAffinityWeight *int32 `json:"hardPodAffinityWeight,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeLabelArgs holds arguments that used to configure the NodeLabel plugin.
+type NodeLabelArgs struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// PresentLabels should be present for the node to be considered a fit for hosting the pod
+	// +listType=atomic
+	PresentLabels []string `json:"presentLabels,omitempty"`
+	// AbsentLabels should be absent for the node to be considered a fit for hosting the pod
+	// +listType=atomic
+	AbsentLabels []string `json:"absentLabels,omitempty"`
+	// Nodes that have labels in the list will get a higher score.
+	// +listType=atomic
+	PresentLabelsPreference []string `json:"presentLabelsPreference,omitempty"`
+	// Nodes that don't have labels in the list will get a higher score.
+	// +listType=atomic
+	AbsentLabelsPreference []string `json:"absentLabelsPreference,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeResourcesFitArgs holds arguments used to configure the NodeResourcesFit plugin.
+type NodeResourcesFitArgs struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// IgnoredResources is the list of resources that NodeResources fit filter
+	// should ignore.
+	// +listType=atomic
+	IgnoredResources []string `json:"ignoredResources,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RequestedToCapacityRatioArgs holds arguments used to configure RequestedToCapacityRatio plugin.
+type RequestedToCapacityRatioArgs struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Points defining priority function shape
+	// +listType=atomic
+	Shape []v1.UtilizationShapePoint `json:"shape"`
```

Comment:
1.18 shipped with this plugin's args referencing the internal `UtilizationShapePoint` type that does not have json tags defined, see https://github.com/kubernetes/kubernetes/pull/88585/files#r405015933

Add a test that exercises this config as described in https://github.com/kubernetes/kubernetes/pull/88585/files#r405015933 and verify its behavior against this PR did not change from 1.18. This may need to use the internal config.UtilizationShapePoint for compatibility and fix this in v1beta1.

Reply:
We can instead copy the types into v1alpha2 without the json tags and add the api-violation exceptions. Right?

Reply:
that should work as well. make sure there's a comment and an issue tracking adding the json tags in v1beta1 so the types don't get blindly copy/pasted